### ISSUE TITLE
Allow using of OsmAnd offline maps

### ISSUE THREAD

**Issue by pejakm** [mvglasow/satstat#88]
The maps are in obf format, like: `Bosnia-herzegovina_europe.obf`. By default, maps are stored in osmand folder.

**Comment by licaon-kter** [comment]
Those used to be on /sdcard/... and user accesible, but now they're in /data/data/.

**Comment by pejakm** [comment]
They're still on /sdcard/osmand on my phone. :) One can select where to download maps.

**Comment by licaon-kter** [comment]
Correct, I had those there and on an app reinstall it stopped using them, looks like I need to move them back. Thanks

**Comment by mvglasow** [comment]
SatStat relies on [Mapsforge](http://www.mapsforge.org) for map display, which currently has no support for the OsmAnd map format. If anyone knows of a way to get Mapsforge to work with OsmAnd files (other than writing a map file reader from scratch – unless anyone volunteers to do that), I'll be happy to implement that.

**Comment by jkufner** [comment]
As far as I know nobody did that yet. I remember some discussions about using Osmand in c:geo (geocaching app which also can use MapForge) few years ago, but nothing happened ever since. Ask on Osmand mailing list, but I believe it is pretty safe to start working on this.

**Comment by mvglasow** [comment]
Well... I currently lack the resources to do this, and since I don't use Osmand, there's little benefit for myself. But if somebody else wants to have a go, I'll happily merge the result.

* The [mapsforge-dev mailing list](https://groups.google.com/forum/#!forum/mapsforge-dev) is a good starting point for any questions related to Mapsforge development.
* As for licensing – SatStat is under GPLv3, so is Osmand. Mapsforge is under LGPLv3 – if your implementation is in any way based on Osmand code, it would inherit the GPL and probably has no chances of getting merged into Mapsforge. But if you develop it as an independent JAR that plugs into Mapsforge, there shouldn't be any licensing issues. Not sure if Mapsforge currently supports "external" readers, but I don't believe the devs would be opposed to extending Mapsforge to make that possible. An app that uses Mapsforge (LGPLv3) with the new Osmand support library (presumably GPLv3) would then have to be under the GPLv3, which is already the case with SatStat.

**Comment by rdoeffinger** [comment]
Out of curiosity, what would be the main purpose of having this support?
Because just having a menu entry to share the current location would provide a way to get the current location shown in OsmAnd, if that's the purpose.
There seems to be a somewhat related feature-request, but it seems to point to the API instead, which I don't think allows embedding the map in another app though: https://github.com/osmandapp/Osmand/issues/3526

**Comment by Rubylion** [comment]
Hey. I'm not in developing, so please don't stone me😃. I use OSMand too, and would love to use the already downloaded (many GBs using up) maps, instead of downloading another ton of GBs from mapforge. As far as I know Openstreet-maps are GPL and used by a variety of GPS-Apps. Wouldn't it be possible to import these as a map overlay?
